### PR TITLE
Make icons use TextProp type for string props and add derives to IconWeight

### PR DESCRIPTION
**Pull request by SleeplessOne1917**
Apologies for the unexpected PR. I'm looking for a good icon library to use with leptos, and the 2 I've tried so far ([leptos-icons](https://github.com/Carlosted/leptos-icons) and [leptos_heroicons](https://github.com/bbstilson/leptos_heroicons)) are not receiving as much active development as this project.

As for the purpose of the PR, the comments in the code changes will make things more clear. I also ran leptosfmt on the project.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -13,7 +13,7 @@ exclude = ["/core"]
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-leptos = "0.5.1"
+leptos = "0.5"
```
Comment (SleeplessOne1917) [inline Cargo.toml:16]: Let me know if you prefer the more specific 3 digit version number.

File: src/icons/address_book.rs [hunk 2]
```diff
@@ -1,38 +1,39 @@
...
 #[component]
 pub fn AddressBook(
-    #[prop(into, default = MaybeSignal::Static(IconWeight::Regular))] weight: MaybeSignal<IconWeight>,
-    #[prop(into, default = MaybeSignal::Static("1em".to_string()))] size: MaybeSignal<String>,
-    #[prop(into, default = MaybeSignal::Static("currentColor".to_string()))] color: MaybeSignal<String>,
-    #[prop(into, default = MaybeSignal::Static(false))] mirrored: MaybeSignal<bool>
+    #[prop(into, default = MaybeSignal::Static(IconWeight::Regular))] weight: MaybeSignal<
+        IconWeight,
+    >,
+    #[prop(into, default = TextProp::from("1em"))] size: TextProp,
```
Comment (SleeplessOne1917) [inline src/icons/address_book.rs:11]: [`TextProp`](https://docs.rs/leptos/latest/leptos/struct.TextProp.html) is a bit more versatile than `MaybeSignal<String>` while also preserving performance.

File: src/lib.rs [hunk 3]
```diff
@@ -1,7 +1,12 @@
 mod icons;
 pub use icons::*;
-            
-#[derive(Clone)]
+
+#[derive(Clone, Copy, PartialEq, Eq)]
```
Comment (SleeplessOne1917) [inline src/lib.rs]: I added:
- `Copy`: this enum doesn't need to be heap allocated, so copy should make it less expensive than cloning.
- `PartialEq, Eq`: makes enum values comparable with equality operator.

File: src/icons/address_book.rs (hunk 2 above)
Comment (SorenHolstHansen) [reply src/icons/address_book.rs:11]: Cool, didn't know that actually

File: Cargo.toml (hunk 1 above)
Comment (SorenHolstHansen) [reply Cargo.toml:16]: This is fine

File: src/lib.rs (hunk 3 above)
Comment (SorenHolstHansen) [reply src/lib.rs]: Nice!

**Review by SorenHolstHansen — APPROVED** [review]
Looks great! Thanks for the PR! I will keep it open a bit longer just to remind myself to publish a new version when I get home. I really should do releases in a github action

File: src/lib.rs (hunk 3 above)
Comment (SleeplessOne1917) [reply src/lib.rs]: I'm making it derive [`Debug`](https://doc.rust-lang.org/std/fmt/trait.Debug.html) as well.

**Comment by SorenHolstHansen** [comment]
@SleeplessOne1917 Thanks a lot for the PR, I have published a new version now

**Pull request merged**